### PR TITLE
T38131 api.models: fix node state validation

### PR DESCRIPTION
**Pull request by JenySadadia**
Need to handle a case where node update operation
do not change the node state.

Fixes: 1f96e91 ("api.models: implement `Node.validate_node_transition`")
Signed-off-by: Jeny Sadadia <jeny.sadadia@collabora.com>

### CODE REVIEW

File: api/main.py [hunk 1]
```diff
@@ -286,16 +286,22 @@ async def put_node(node_id: str, node: Node, token: str = Depends(get_user)):
     """Update an already added node"""
     try:
         node.id = ObjectId(node_id)
-        node_from_id = await get_node(node.id)
-        is_valid, message = node_from_id.validate_node_transition(node.state)
+        node_from_id = await db.find_by_id(Node, node_id)
+        if not node_from_id:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail=f"Node not found with id: {node.id}"
```
Comment (mgalka) [inline api/main.py:293]: I think we should use here HTTP 404. Even the detailed message says that node hasn't been found.

File: api/models.py [hunk 2]
```diff
@@ -239,8 +239,10 @@ def translate_fields(cls, params: dict):
             translated['parent'] = ObjectId(parent)
         return translated
 
-    def validate_node_transition(self, new_state):
+    def validate_node_state_transition(self, new_state):
         """Validate Node.state transitions"""
+        if new_state == self.state:
+            return True, "No validation is required"
```
Comment (mgalka) [inline api/models.py:245]: Maybe it'd be better to give more informative message like: 
```python
f"Transition to the same state: { new_state }. No validation required."
```

**Review by mgalka — COMMENTED** [review]
minor remarks

File: api/main.py (hunk 1 above)
Comment (JenySadadia) [reply api/main.py:293]: Definitely. It was a blunder.

**Review by mgalka — APPROVED** [review]
LGTM

**Pull request merged**